state/actions/fetchFiles.ts

```diff
@@ -1,117 +1,94 @@
-            content: res.data.files?.[filename]?.content || "",
-          }));
-          // Sort files so that the source files are first
-          // In case of other files leave the order as it its
-          files.sort((a, b) => {
-            const aBasename = a.name.split('.')?.[0];
-            const aCext = a.name?.toLowerCase().endsWith('.c');
-            const bBasename = b.name.split('.')?.[0];
-            const bCext = b.name?.toLowerCase().endsWith('.c');
-            // If a has c extension and b doesn't move a up
-            if (aCext && !bCext) {
-              return -1;
-            }
-            if (!aCext && bCext) {
-              return 1
-            }
-            // Otherwise fallback to default sorting based on basename
-            if (aBasename > bBasename) {
-              return 1;
-            }
-            if (bBasename > aBasename) {
-              return -1;
-            }
-            return 0;
-          })
-          state.loading = false;
-          if (files.length > 0) {
-            state.logs.push({
-              type: "success",
-              message: "Fetched successfully ✅",
-            });
-            state.files = files;
-            state.gistId = gistId;
-            state.gistName = Object.keys(res.data.files)?.[0] || "untitled";
-            state.gistOwner = res.data.owner?.login;
-            return;
-          } else {
-            // Open main modal if now files
-            state.mainModalOpen = true;
-          }
-          return Router.push({ pathname: "/develop" });
-        }
-        state.loading = false;
-      })
-      .catch((err) => {
-        // console.error(err)
-        state.loading = false;
-        state.logs.push({
-          type: "error",
-          message: `Couldn't find Gist with id: ${gistId}`,
-        });
-        return;
-      });
-    return;
+      const files = {
+        ...res.data.files,
+        ...headerFiles
+      };
+      res.data.files = files;
+    }
+
+    if (!res.data.files) throw Error("No files could be fetched from given gist id!")
+
+    const files: IFile[] = Object.keys(res.data.files).map((filename) => ({
+      name: res.data.files?.[filename]?.filename || "untitled.c",
+      language: res.data.files?.[filename]?.language?.toLowerCase() || "",
+      content: res.data.files?.[filename]?.content || "",
+    }));
+
+    // Sort files so that the source files are first
+    // In case of other files leave the order as it its
+    files.sort((a, b) => {
+      const aBasename = a.name.split('.')?.[0];
+      const aExt = a.name.split('.').pop() || '';
+      const bBasename = b.name.split('.')?.[0];
+      const bExt = b.name.split('.').pop() || '';
+
+      // default priority is undefined == 0
+      const extPriority: Record<string, number> = {
+        c: 3,
+        md: 2,
+        h: -1
+      }
+
+      if (extPriority[aExt] || extPriority[bExt]) {
```

Comment:
The condition isn't really needed, right? comp can be computed unconditionally, and it will be 0 in case both extensions are unknown...